src/components/Todos/Todos.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { Todo } from '../../types/Todo';
+import { TempTodo } from '../TempTodo';
+
+type Props = {
+  todos: Todo[] | null;
+  newTodoTitle: string;
+  isTodoAdded: boolean;
+  removeTodo: (targetId: number) => void;
+  targetTodosIds: number[];
+  changeTodoStatus: (todoId: number, status: boolean) => void;
+  callEditeMode: (title: string) => void;
+  updatingTitle: string;
+  setUdatingTitle: (value: string) => void;
+  submitTitleUpdating: (id: number, oldTitle: string) => void;
+};
+
+export const Todos: React.FC<Props> = ({
+  todos,
+  newTodoTitle,
+  isTodoAdded,
+  removeTodo,
+  targetTodosIds,
+  changeTodoStatus,
+  callEditeMode,
+  updatingTitle,
+  setUdatingTitle,
+  submitTitleUpdating,
+}) => {
+  const [editedTodoId, setEditedTodoId] = useState(0);
+
+  return (
+    <TransitionGroup>
+      {todos?.map(todo => (
+        <CSSTransition
+          key={todo.id}
+          timeout={300}
+          classNames="item"
+        >
+          <div
+            id={`${todo.id}`}
+            className={classNames('todo', { completed: todo.completed })}
+          >
+            <label className="todo__status-label">
+              <input
+                type="checkbox"
+                className="todo__status"
+                checked={todo.completed}
+                onChange={() => changeTodoStatus(todo.id, !todo.completed)}
+              />
+            </label>
+
+            {editedTodoId === todo.id ? (
+              <form>
+                <input
+                  type="text"
+                  className="todo__title-field"
+                  placeholder="Empty todo will be deleted"
+                  value={updatingTitle}
+                  onChange={(event) => setUdatingTitle(event.target.value)}
+                  onKeyDown={(e) => {
```

Comment:
Please create separate functions for all events. Firstly, you trash jsx, and secondly, these functions are anonymous and they are recreated on each re-render